api/datasets.go

```diff
@@ -227,9 +227,19 @@ func (h *DatasetHandlers) listHandler(w http.ResponseWriter, r *http.Request) {
 	res := []dsref.VersionInfo{}
 	var err error
 	if params.Raw {
-		resRaw, err = h.ListRawRefs(r.Context(), params)
+		got, _, err := h.inst.Dispatch(r.Context(), "dataset.listrawrefs", params)
+		ok := false
+		resRaw, ok = got.(string)
+		if err != nil || !ok {
```

Comment:
If `err` is nil, but `ok` is false, then conditional will be executed with a nil err, which won't do anything helpful. The state of `ok` needs to be checked separately.

There's a few places this is happening, please fix.